Bring clarity to PWA manifest requirements needed to use getInstalledRelatedApps

ishika-patel opened the pull request:
<!-- 🙌 Thanks for contributing to MDN Web Docs. 🙌

Add details below to help us review your pull request (PR).
Explain your changes and link to a related issue or pull request.
Your PR may be delayed or closed if you don't provide enough information. -->

### Description

This PR clarifies how a web developer can set up the web app manifest for their PWA to leverage getInstalledRelatedApps. This is done by providing an example of the PWAs self identifying related_applications manifest field with the required properties and updating the related_applications manifest member page to include `webapp` as a viable option.

### Motivation

<!-- ❓ Why are you making these changes and how do they help readers? -->
getInstalledRelatedApps for PWAs has recently shipped in desktop Chrome and Edge. This change will give web developers clarity on what is required of in the setup of their web app manifest to leverage this API.

ishika-patel committed:
Add example for related_applications field to use getInstalledRelatedApps

This PR clarifies how a web developer can set up the web app manifest for their PWA to leverage getInstalledRelatedApps. This is done by providing an example of the PWAs self identifying related_applications manifest field.

ishika-patel committed:
Update subtitle for getInstalledRelatedApps Example

liminzhu commented on files/en-us/web/api/navigator/getinstalledrelatedapps/index.md:
```diff
@@ -32,6 +32,23 @@ Defining the relationship is done in a different way depending on the type of ap
   - A self-defining entry inside its own `related_applications` manifest member in the case of a PWA checking if it is installed on the underlying platform.
```
Recommend putting the id requirement here. I think having an example is nice, but if there is requirement for id, we should convey it here and have the example as a supplement (as opposed to conveying it in the example).

ishika-patel replied:
The requirement for gIRA on PWAs is to have id. However, this is an optional field w.r.t. related_applications in general.

Will update and add the requirement in the Defining relationship section above.

ishika-patel committed:
Update related_applications page

This commit updates the related_applications page to include webapps and addresses PR feedback.

ishika-patel committed:
Merge branch 'mdn:main' into gIRA

ishika-patel committed:
Update index.md

Fix syntax for code examples in related_applications

ishika-patel committed:
Merge branch 'gIRA' of https://github.com/ishika-patel/content into gIRA

captainbrosset commented on files/en-us/web/progressive_web_apps/manifest/reference/related_applications/index.md:
```diff
@@ -10,8 +10,8 @@ sidebar: pwasidebar
 
-The `related_applications` manifest member is used to specify one or more native applications that are related to your web application.
-It can be used with the [`prefer_related_applications`](/en-US/docs/Web/Progressive_web_apps/Manifest/Reference/prefer_related_applications) manifest member, which indicates a preference for installing either a related native application or your web application.
+The `related_applications` manifest member is used to specify one or more applications that are related to your web application. These can include native applications and even to self-define the web application to leverage web APIs.
```
I was having a bit of trouble parsing this section, so here is a suggestion. Let me know if this helps:

```suggestion
The `related_applications` manifest member is used to specify one or more applications that are related to your web application, such as platform-specific applications (also known as native applications) or installed Progressive Web Apps.
```

captainbrosset commented on files/en-us/web/progressive_web_apps/manifest/reference/related_applications/index.md:
```diff
@@ -10,8 +10,8 @@ sidebar: pwasidebar
-The `related_applications` manifest member is used to specify one or more native applications that are related to your web application.
-It can be used with the [`prefer_related_applications`](/en-US/docs/Web/Progressive_web_apps/Manifest/Reference/prefer_related_applications) manifest member, which indicates a preference for installing either a related native application or your web application.
+The `related_applications` manifest member is used to specify one or more applications that are related to your web application. These can include native applications and even to self-define the web application to leverage web APIs.
+It can also be used with the [`prefer_related_applications`](/en-US/docs/Web/Progressive_web_apps/Manifest/Reference/prefer_related_applications) manifest member, which indicates a preference for installing either a related native application or your web application.
```
"It" can be confusing at times. Better to repeat what we're really talking about (I know you didn't change this in your PR, so this is a drive-by suggestion).

```suggestion
The `related_applications` manifest member can also be used with the [`prefer_related_applications`](/en-US/docs/Web/Progressive_web_apps/Manifest/Reference/prefer_related_applications) manifest member, which indicates a preference for installing either a related native application or your web application.
```

ishika-patel replied:
Accepted this suggestion with the most recent commit.

captainbrosset commented on files/en-us/web/progressive_web_apps/manifest/reference/related_applications/index.md:
```diff
@@ -45,15 +45,23 @@ It can be used with the [`prefer_related_applications`](/en-US/docs/Web/Progress
   }
 ]
+
+/* Self-defining web application on one platform specified only by id */
```
Not sure if better, but I thought it needed a little bit more explanation of what we mean by "self-defining". Feel free to reject/change.

```suggestion
/* Related web application specified by id, self-defined by its own web app manifest. */
```

ishika-patel replied:
Accepted this suggestion with the most recent commit.

captainbrosset commented on files/en-us/web/progressive_web_apps/manifest/reference/related_applications/index.md:
```diff
@@ -45,15 +45,23 @@ It can be used with the [`prefer_related_applications`](/en-US/docs/Web/Progress
     - `platform`
       - : A string that identifies the platform on which the application can be found.
-        Examples include `amazon` (Amazon App Store), `play` (Google Play Store), and `windows` (Windows Store).
+        Examples include `amazon` (Amazon App Store), `play` (Google Play Store), `windows` (Windows Store), and `webapp` (self-define your PWA in its own manifest).
```
```suggestion
        Examples include `amazon` (Amazon App Store), `play` (Google Play Store), `windows` (Windows Store), and `webapp` (for installed Progressive Web Apps).
```

ishika-patel replied:
Accepted this suggestion with the most recent commit.

captainbrosset commented on files/en-us/web/progressive_web_apps/manifest/reference/related_applications/index.md:
```diff
@@ -64,20 +72,19 @@ It can be used with the [`prefer_related_applications`](/en-US/docs/Web/Progress
 ## Description
 
-A "related application" is a {{Glossary("native")}} application that provides functionality similar to your web app, often with additional features or better integration with users' devices.
+A "related application" is the web app itself or {{Glossary("native")}} application that provides functionality similar to your web app, often with additional features or better integration with users' devices.
```
```suggestion
A "related application" is the web app itself, when installed as a Progressive Web App (PWA), or a {{Glossary("native")}} application that provides functionality similar to your web app, often with additional features or better integration with users' devices.
```

captainbrosset commented on files/en-us/web/progressive_web_apps/manifest/reference/related_applications/index.md:
```diff
@@ -136,6 +143,21 @@ If you want to indicate to browsers that you prefer users to be given the option
 
+### Specifying a related web application
+
+This example shows how to specify the self-defining entry for PWAs in your web app's manifest file.
```
I felt like this need a few more words to clarify the intent.

```suggestion
If your web app can be installed as a Progressive Web App (PWA) on the device, for example to take advantage of features which integrate your PWA into the operating system, you can self-reference your web app in the manifest file:
```

captainbrosset reviewed (approved):
This looks good to me in terms of scope of content. I did make a few reword suggestions.
I think this is good to turn into an actual PR (remove the draft state) and get an MDN reviewer to take a look.

captainbrosset committed:
Merge branch 'main' into gIRA

wbamberg committed:
Merge branch 'main' into gIRA

caugner committed:
Merge branch 'main' into gIRA

ishika-patel committed:
Add Suggested Clarifications

This commit addresses open suggestions on the PR made to clarify the related_applications page and makes one grammar change in the gIRA page.

ishika-patel committed:
Merge branch 'gIRA' of https://github.com/ishika-patel/content into gIRA

ishika-patel committed:
Merge branch 'main' into gIRA

wbamberg reviewed (commented):
Thanks for the PR!

I must say I found this quite hard to understand and think it would benefit a lot from a how-to guide laying out the use cases for the API+manifest key. But I don't think this is your fault - it's a pre-existing problem in the docs - and I think overall your PR is definitely an improvement.

wbamberg commented:
Reading these pages, I wondered about making a new [how-to](https://developer.mozilla.org/en-US/docs/Web/Progressive_web_apps/How_to) guide on getting related apps. It seems to me that the current `getInstalledRelatedApps()` page is trying to pack a lot of exposition into a reference page. A how-to might be better able to express the use case for it and have the space to explain the rather complicated setup (which isn't directly related to the API itself).

I don't think this would be needed for this PR but I think it would help. @ishika-patel , I also don't know if you would have the time to do this: if not I might give it a go myself.

ishika-patel committed:
Update files/en-us/web/progressive_web_apps/manifest/reference/related_applications/index.md

Add suggestions from reviewer.

Co-authored-by: wbamberg <will@bootbonnet.ca>

ishika-patel committed:
Update files/en-us/web/progressive_web_apps/manifest/reference/related_applications/index.md

Add reviewer suggestion.

Co-authored-by: wbamberg <will@bootbonnet.ca>

ishika-patel committed:
Update files/en-us/web/progressive_web_apps/manifest/reference/related_applications/index.md

Apply reviewer suggestions.

Co-authored-by: wbamberg <will@bootbonnet.ca>

ishika-patel commented on files/en-us/web/progressive_web_apps/manifest/reference/related_applications/index.md:
```diff
@@ -45,15 +45,23 @@ It can be used with the [`prefer_related_applications`](/en-US/docs/Web/Progress
     - `platform`
       - : A string that identifies the platform on which the application can be found.
-        Examples include `amazon` (Amazon App Store), `play` (Google Play Store), and `windows` (Windows Store).
+        Examples include `amazon` (Amazon App Store), `play` (Google Play Store), `windows` (Windows Store), and `webapp` (for installed Progressive Web Apps).
```
@wbamberg This is a slight reference to the web app being installed as a PWA, but I agree that "installed" in this case is misleading. Will make an update to generalize to only Progressive Web Apps

ishika-patel commented on files/en-us/web/progressive_web_apps/manifest/reference/related_applications/index.md:
```diff
@@ -148,4 +170,4 @@ If you want to indicate to browsers that you prefer users to be given the option
 - [`prefer_related_applications`](/en-US/docs/Web/Progressive_web_apps/Manifest/Reference/prefer_related_applications) manifest member
 - [The web app manifest](/en-US/docs/Web/Progressive_web_apps/Guides/Making_PWAs_installable#the_web_app_manifest) for making your web app installable
-- {{domxref("Navigator.getInstalledRelatedApps()")}}
+- {{domxref("Navigator.getInstalledRelatedApps()")}} for checking if your web app is installed. This requires specification of the unique `id` field in this manifest member.
```
Correct, I see the disconnect. I was refreaming around getInstalledRelatedApps for PWAs as that was top of my mind. Will update the description. Great catch!

ishika-patel committed:
Merge branch 'mdn:main' into gIRA

ishika-patel commented on files/en-us/web/progressive_web_apps/manifest/reference/related_applications/index.md:
```diff
@@ -10,8 +10,8 @@ sidebar: pwasidebar
-The `related_applications` manifest member is used to specify one or more native applications that are related to your web application.
-It can be used with the [`prefer_related_applications`](/en-US/docs/Web/Progressive_web_apps/Manifest/Reference/prefer_related_applications) manifest member, which indicates a preference for installing either a related native application or your web application.
+The `related_applications` manifest member is used to specify one or more applications that are related to your web application, such as platform-specific applications (also known as native applications) or installed Progressive Web Apps.
+The `related_applications` manifest member can also be used with the [`prefer_related_applications`](/en-US/docs/Web/Progressive_web_apps/Manifest/Reference/prefer_related_applications) manifest member, which indicates a preference for installing either a related native application or your web application.
```
Correct, one example would be the Outlook Native App and the Outlook PWA. For use of the webapp flavor the expectation is that it more or less the app itself. getInstalledRelatedApps for PWAs, which uses the webapp related_applications flavor, must be requested on the same domain, and within the scope of your PWA.

ishika-patel committed:
Update related_applications

Change phrasing for items that are too gIRA and PWA specific in the related_applications page based on reviewer suggestions.

ishika-patel committed:
Modify reference of native

Use the set reference to "platform-specific" not "native"

wbamberg reviewed (approved):
👍 thank you @ishika-patel !

ishika-patel commented:
@wbamberg Thank you for your review! I currently do not have the capacity to take on the "how-to" guide. Feel free to give it a go!

wbamberg commented:
> @wbamberg Thank you for your review! I currently do not have the capacity to take on the "how-to" guide. Feel free to give it a go!

OK, thanks for letting me know!
